api/v1alpha4/common_types.go

```diff
@@ -53,6 +53,12 @@ const (
 	// on the reconciled object.
 	PausedAnnotation = "cluster.x-k8s.io/paused"
 
+	// WatchLabel is a label othat can be applied to any Cluster API object.
+	//
+	// Controllers which allow for selective reconciliation may check this label and proceed
+	// with reconciliation of the object only if this label and a configured value is present.
+	WatchLabel = "cluster.x-k8s.io/watch"
```

Comment:
I struggled with naming here - this seemed most consistent with the already existing labels.

Reply:
I have no brilliant suggestions, but I would like to have something like "watch-filter" or "watch-selector" to make it clear that this value will the used to filter out / to selected

Reply:
I changed it to `watch-filter` and follows that naming - WDYT?